### PR TITLE
Joda time config converters

### PR DESCRIPTION
**Pull request by queimadus**
Add `ConfigConvert`ers to `joda-time`'s `Instant`, `Period`, `Interval`, `Duration`, `DateTimeFormatter` and `DateTimeZone`.

Brought to you by [scala-swarm scala hackathon](https://www.scala-swarm.org/), courtesy of @queimadus and @andrebeat.

### CODE REVIEW

**Review by melrief — APPROVED** [review]
Good stuff! Thank you very much for this PR. Waiting for another review (maybe @leifwickland ?) and then merging.

I'd like to add everyone involved to the contributors of Pureconfig. If you guys agree, please create another PR that adds your names/emails/nicknames to the [CONTRIBUTORS](https://github.com/pureconfig/pureconfig/blob/master/CONTRIBUTORS) file.

**Comment by derekmorr** [comment]
This looks great. Thanks for doing this!

My only question is should we update `modules/joda/src/main/tut/README.md` to list what Joda types are supported?

File: modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala
Comment (leifwickland) [inline modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala:77]: Nitpick: please move the tests for types which are not configurable to a `pureconfig.module.joda.JodaSpec` file.

File: modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala
Comment (leifwickland) [inline modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala:178]: Nitpick: s/Marshelling/Marshalling/

File: modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala
Comment (leifwickland) [inline modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala:165]: Suggestion: Comparing the intervals and put the lesser in _1 and greater in _2. That will generate valid cases twice as fast.

File: modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala
Comment (leifwickland) [inline modules/joda/src/test/scala/pureconfig/module/joda/configurable/ConfigurableSuite.scala:178]: Did you try generating a `Long` and constructing a `Period` using the single-argument constructor? I expect that may give you more consistent results with broader coverage.

**Review by leifwickland — COMMENTED** [review]
Thanks, @queimadus! I made some suggestions. Please consider updating the PR.

**Pull request merged**
